### PR TITLE
Running preview

### PR DESCRIPTION
**Pull request by coolo**

### CODE REVIEW

**Comment by coolo** [comment]
Both Ludwig and Richard tested the prototype and had serious trouble with it. So I guess having the live view in an extra tab is the easier to understand concept

File: assets/javascripts/running.js [hunk 1]
```diff
@@ -263,6 +264,7 @@ function initLivestream() {
     }
     setResolution(livestream_size);
     sel_resolution.value = livestream_size;
+    */
```
Comment (okurz) [inline assets/javascripts/running.js]: dead code

File: assets/javascripts/running.js [hunk 2]
```diff
@@ -295,6 +298,7 @@ function setupRunning() {
     $('#sel_resolution').change(function() {
         setResolution($(this).val());
     });
+    */
```
Comment (okurz) [inline assets/javascripts/running.js]: again

File: lib/OpenQA/WebAPI/Controller/Test.pm [hunk 3]
```diff
@@ -257,8 +258,9 @@ sub show {
         $self->stash(worker => $job->worker);
         $self->stash(job    => $job);
         $self->stash('backend_info', decode_json($job->backend_info || '{}'));
-        $self->render('test/running');
-        return;
+        $self->stash(running => 1);
+        #$self->render('test/running');
+        #return;
```
Comment (okurz) [inline lib/OpenQA/WebAPI/Controller/Test.pm]: dead code

**Comment by okurz** [comment]
:angry:

File: assets/javascripts/running.js (hunk 1 above)
Comment (coolo) [inline assets/javascripts/running.js]: really? what does "quick and dirty" mean to *you* ? :)

**Comment by coolo** [comment]
I moved the live image into its own tab:

image

TODO: readd interactive buttons and add some kind of information where the test is (i.e. show running module within that tab)

**Comment by coolo** [comment]
more TODO: live update of the details tab

**Comment by coolo** [comment]
TODO: find a place for the worker name and VNC port

**Comment by aaannz** [comment]
Hmm.. didn't try this yet, but judging from the screenshot I miss the list of modules.

File: assets/javascripts/running.js (hunk 1 above)
Comment (okurz) [inline assets/javascripts/running.js]: It means asdil12 can disable code and you split the commit into "stuff to push in a PR" and "stuff to keep for later in my local WIP commits".

**Comment by coolo** [comment]
Readding the necessary infos shows some conflict within the layout - but at least we should have all functionality back

image

**Comment by aaannz** [comment]
I like the idea where it is going, so LGTM, but judging the last screenshot the infobox looks like from last century :)
Next, details tab is not regularly updated when new module is being tested as it was in the old view. So switching liveview<->details does not present coherent picture.

**Comment by lnussel** [comment]
real bug: if the job goes to waiting and one refreshes the page the live view tag is gone

**Comment by lnussel** [comment]
one annoyance that was already present in the old code is that if one saves a needle it offers to restart the job instead of reload_needles_and_retry

**Comment by coolo** [comment]
ok, I can have a look at the waiting state tomorrow. I already noticed that some places check and some don't. 

But everything else you mention is beyond this PR

**Comment by coolo** [comment]
Tested 5c57fb7 on tortuga

**Comment by coolo** [comment]
#live is life on openqa.opensuse.org - https://www.youtube.com/watch?v=xtn2LQctSTE

**Pull request merged**